ocaml/src/nsm_model.ml

```diff
@@ -32,6 +32,23 @@ module OsdInfo = struct
   type ip = string [@@deriving show, yojson]
   type port = int [@@deriving show, yojson]
 
+  type endpoint_uri = string [@@deriving show, yojson]
+
+  type host = string
+  type endpoint = Net_fd.transport * host * port
+
+  let parse_endpoint_uri s =
+    let uri = Uri.of_string s in
+    let transport = match Uri.scheme uri with
+      | None -> failwith "Transport should be 'tcp' or 'rdma'"
+      | Some "tcp" -> Net_fd.TCP
```

Comment:
I don't know of any automatic conversion that it does ... but the help message mentions 'tcp' and 'rdma' with lower case, so I don't see a problem.
Or are you trying to say it should support upper case schemes? (I don't see why we should.)